tests/unit/s2n_server_key_share_extension_test.c

```diff
@@ -253,8 +265,10 @@ int main(int argc, char **argv)
 
     /* Test Shared Key Generation */
     {
+        const struct s2n_ecc_preferences *ecc_pref = NULL;
         int shared_secret_size[3] = { 32, 48, 32 };
-        for (int i = 0; i < s2n_ecc_evp_supported_curves_list_len; i++)
+        int i = 0;
+        do
```

Comment:
Why the do while instead of the for?

Reply:
Prior, we were obtaining the `s2n_ecc_evp_supported_curves_list_len` to iterate on from a hard coded static list of supported curves, now that we are setting the supported curves on the config, we need to obtain this list after the connection is set, which is within the loop. Hence, converted the for to a do-while.